server/sonar-server/src/main/java/org/sonar/server/computation/ws/ActivityWsAction.java

```diff
@@ -36,15 +38,22 @@
 import org.sonar.db.ce.CeActivityDto;
 import org.sonar.db.ce.CeActivityQuery;
 import org.sonar.db.ce.CeTaskTypes;
+import org.sonar.db.component.ComponentDto;
+import org.sonar.db.component.ComponentDtoFunctions;
+import org.sonar.db.component.ComponentQuery;
+import org.sonar.server.exceptions.BadRequestException;
 import org.sonar.server.exceptions.ForbiddenException;
 import org.sonar.server.user.UserSession;
 import org.sonar.server.ws.WsUtils;
 import org.sonarqube.ws.Common;
 import org.sonarqube.ws.WsCe;
 
+import static java.lang.String.format;
+
 public class ActivityWsAction implements CeWsAction {
 
   private static final String PARAM_COMPONENT_UUID = "componentId";
+  private static final String PARAM_COMPONENT_QUERY = "componentQuery";
```

Comment:
I'm not sure the parameter should be generic (`q`), as it's restricted to components. Moreover description is specific to this WS.